pkg/broker/config/volume/volume.go

```diff
@@ -72,27 +72,23 @@ func NewTargetsFromFile(opts ...Option) (config.ReadOnlyTargets, error) {
 		return nil, err
 	}
 
-	initialized := make(chan struct{}, 1)
-	go t.watchWith(watcher, initialized)
-	// Block until we know for sure the events handler is properly set up.
-	<-initialized
+	t.watchWith(watcher)
 	return t, nil
 }
 
-func (t *Targets) watchWith(watcher *fsnotify.Watcher, initialized chan struct{}) {
-	defer watcher.Close()
-	done := make(chan struct{}, 1)
+func (t *Targets) watchWith(watcher *fsnotify.Watcher) {
 	configFile := filepath.Clean(t.path)
 	configDir, _ := filepath.Split(t.path)
 	realConfigFile, _ := filepath.EvalSymlinks(t.path)
+	watcher.Add(configDir)
+
 	go func() {
 		for {
 			select {
 			case event, ok := <-watcher.Events:
 				if !ok {
 					// 'Events' channel is closed.
-					done <- struct{}{}
-					return
+					break
```

Comment:
you'll actually need a named break statement here (or do the return with the defer inside the function) since just doing the bear break just breaks you out of the select

Reply:
ah, right! No wonder it gave me warning.